### ISSUE TITLE
Adjust reading and writing data guides regarding ManyToMany issues

### ISSUE THREAD

**Issue by PaddyS** [shopware/docs#41]
https://github.com/shopware/docs/pull/19#issuecomment-763887436

> I like the comparison to show the pitfall. The scenario up front reads a bit colloquial but not totally made up. I just think it is not nice from the docs to tell the reader they "messed" up.
There is an other pitfall I like to have mentioned in the docs somehow. The product_category.repository can't be used to search as it is a MappingEntityDefinition and it will crash. From my POV I'd say: I can just iterate product_category and look for my entry/search for my entry and just update the category id. I don't want to state that you should say: you can't iterate over mapping entities by design. This can be a bit disappointing (hit me already). But you can mention that you can't update it e.g. by the following reason: When you call update you pass the primary key(s). In that case you have two. You can't update primary keys and thus you can't update the entry but have to delete and create it. Additionally you can suggest to iterate the categories from the product entity so they don't think of my iteration case.

TL;DR:
Mention that mapping entities can neither be read, nor written / updated.
